### PR TITLE
ci: add configuration fileds to codecov.yaml

### LINKED ISSUE #6667: Add flags when upload test coverage to codecov.io
## Development Task

- [ ] add flags management in `codecov.yaml` to separate unit and integration testing coverage.
- [ ] ignore the `tests` dir, it's for integration tests.

Ref:
- https://github.com/PingCAP-QE/ci/issues/2171

### PR DESCRIPTION
**Pull request by wuhuizuo**
### What problem does this PR solve?

Ref: 
- https://github.com/PingCAP-QE/ci/issues/2171

Issue Number: Close #6667

### What is changed and how does it work?

```commit-message
ci: add configuration fileds to codecov.yaml

- `ignore`: ignore integration test cases or tools paths.
- `flags`: to split static for unit and integration testing.
- `comment`: add `flags` section in comments.
```

### Check List

Tests

- No code

Code changes

- None

Side effects

- None

Related changes

- None

### Release note

```release-note
None.
```

### CODE REVIEW

File: codecov.yml [hunk 1]
```diff
@@ -10,3 +10,20 @@ coverage:
...
+  layout: "header, diff, flags"
+  behavior: default
+  require_changes: false
+
+flag_management:
+  default_rules: # the rules that will be followed for any flag added, generally
+    carryforward: true
+    statuses:
+      - type: project
+        target: 85%
```
Comment (wuhuizuo) [inline codecov.yml:24]: 70%

Comment (wuhuizuo) [inline codecov.yml:24]:
```suggestion
        target: 70%
```

File: codecov.yml [hunk 2]
```diff
@@ -10,3 +10,20 @@ coverage:
...
+      - type: patch
+        target: 85%
```
Comment (wuhuizuo) [inline codecov.yml:26]:
```suggestion
        target: 70%
```

**Review by nolouch — APPROVED** [review]

File: codecov.yml [hunk 3]
```diff
@@ -10,3 +10,20 @@ coverage:
...
+  default_rules: # the rules that will be followed for any flag added, generally
+    carryforward: true
+    statuses:
+      - type: project
+        target: 70%
+      - type: patch
+        target: 70%
+
+ignore:
+  - tests/** # integration test cases or tools.
```
Comment (nolouch) [inline codecov.yml:29]: Does ignoring here will reduce the test coverage?

Comment (wuhuizuo) [reply codecov.yml:29]: Maybe, but they are test tools or integration test cases, should be ignored.

**Review by purelind — APPROVED** [review]
/lgtm

File: codecov.yml (hunk 2 above)
Comment (HuSharp) [inline codecov.yml:26]: I noticed ci failed too much, is it too high?
image

**Pull request merged**